.github/workflows/run-tests.yml

```
@@ -29,7 +44,9 @@ env:
   BUILDKITE_ANALYTICS_TOKEN: ${{ secrets.BUILDKITE_ANALYTICS_TOKEN }}
 
 jobs:
+
   misc-checks:
+    if: ${{ inputs.run_single_functional_test != true }}
```

You need to quote the boolean values.

```suggestion
    if: ${{ github.event.inputs.run_single_functional_test != 'true' }}
```

---

I tried it with quoting the boolean value and it didn't work, whereas what is currently in the PR works. Not sure why, but I've tried both ways multiple times and I'm very sure.

---

hm.. actually, I need to test this more

---

I don't think `github.events.inputs` is required though, per https://github.blog/changelog/2022-06-10-github-actions-inputs-unified-across-manual-and-reusable-workflows/

---

ok yeah, it has to be `!= true`

---

https://github.com/temporalio/temporal/actions/runs/9961930639 this run is with the single_test input boolean checked, and `!= 'true'`, and the unit tests and integration tests run, which is not what I want

---

https://github.com/temporalio/temporal/actions/runs/9962244076 this run is with the single_test input boolean checked and `!= true`, and the unit tests and integration tests are correctly skipped

---

Just to be clear, did you use single quote?
`'true'` works
`"true"` does not work